File: 206 Reverse Linked List/206 Reverse Linked List.js
var reverseList = function(head){
    var cp = head, np = null, pp;
    while(cp!=null){
        //獲取下一個node
        pp = cp.next;
        
        //設定當前node的下一個對應node為上一個
        cp.next = np;
        
        //指標切換
        np = cp;
        cp = pp;
    }
    return np;
}
//Min:132ms

File: 206 Reverse Linked List/206 Reverse Linked List.c
struct ListNode* reverseList(struct ListNode* head){
    struct ListNode *cp = head, *np = NULL, *pp;
    while(cp!=NULL){
        pp = cp->next;
        cp->next = np;
        np = cp;
        cp = pp;
    }
    return np;
}
//Min:0ms